apps/heft/src/utilities/subprocess/ScopedLoggerManager.ts

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { ITerminalProvider, Terminal } from '@rushstack/node-core-library';
+
+import {
+  ISubprocessMessageBase,
+  ISerializedErrorValue,
+  ISubprocessApiCallArgWithValue
+} from './SubprocessCommunication';
+import {
+  SubprocessCommunicationManagerBase,
+  ISubprocessCommunicationManagerBaseOptions
+} from './SubprocessCommunicationManagerBase';
+import { TerminalProviderManager } from './TerminalProviderManager';
+import { IScopedLogger, ScopedLogger } from '../../pluginFramework/logging/ScopedLogger';
+import { HeftSession } from '../../pluginFramework/HeftSession';
+import { SubprocessRunnerBase } from './SubprocessRunnerBase';
+
+const SCOPED_LOGGER_MANAGER_REQUEST_LOGGER_MESSAGE_TYPE: string = 'scopedLoggerManagerRequestLogger';
+const SCOPED_LOGGER_MANAGER_PROVIDE_LOGGER_MESSAGE_TYPE: string = 'scopedLoggerManagerProvideLogger';
+const SCOPED_LOGGER_EMIT_ERROR_WARNING_MESSAGE_TYPE: string = 'scopedLoggerEmitErrorWarning';
+
+interface IRequestLoggerMessage extends ISubprocessMessageBase {
+  loggerName: string;
+}
+
+interface IProvideLoggerMessage extends ISubprocessMessageBase {
+  loggerName: string;
+  terminalProviderId?: number;
+  error?: ISubprocessApiCallArgWithValue<ISerializedErrorValue>;
+}
+
+interface IEmitErrorOrWarning extends ISubprocessMessageBase {
+  loggerId: number;
+  errorOrWarning: ISubprocessApiCallArgWithValue<ISerializedErrorValue>;
+  isError: boolean;
+}
+
+interface IPromiseResult<TResult> {
+  resolve: (result: TResult) => void;
+  reject: (error: Error) => void;
+}
+
+export interface IScopedLoggerManagerOptions extends ISubprocessCommunicationManagerBaseOptions {
+  terminalProviderManager: TerminalProviderManager;
+  heftSession?: HeftSession;
+}
+
+export class ScopedLoggerManager extends SubprocessCommunicationManagerBase {
```

Comment:
ScopedLoggerManager sounds very similar to the functionality of the LoggingManager class, which also manages ScopedLoggers. Maybe something like SubprocessLoggerManager for this one to communicate that it implements the SubprocessCommunicationManagerBase? Or SubprocessScopedLoggerManager, which is long but accurate, or SubprocessLoggingManager, which follows the same convention as LoggingManager?

Reply:
Let's go with `SubprocessLoggingManager`